src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -9848,7 +9858,8 @@ private bool ScanCast()
             if (type == ScanTypeFlags.PointerOrMultiplication ||
                 type == ScanTypeFlags.NullableType ||
                 type == ScanTypeFlags.MustBeType ||
-                type == ScanTypeFlags.AliasQualifiedName)
+                type == ScanTypeFlags.AliasQualifiedName ||
+                type == ScanTypeFlags.TupleType)
```

Comment:
This is not correct. This causes ((Int32.MaxValue, Int32.MaxValue)) be considered a cast no matter what follows it, even though it is an expression.